cpp/libs/gst_nmos_sender_plugin/gst_nmos_sender_plugin.cpp

```diff
@@ -330,41 +415,49 @@ static void gst_nmossender_class_init(GstNmossenderClass* klass)
 /* Object initialization */
 static void gst_nmossender_init(GstNmossender* self)
 {
-    self->payloader = gst_element_factory_make("rtpvrawpay", "payloader");
-    self->udpsink   = gst_element_factory_make("udpsink", "udpsink");
+    self->queue           = gst_element_factory_make("queue", NULL);
+    self->video_payloader = gst_element_factory_make("rtpvrawpay", NULL);
+    self->audio_payloader = gst_element_factory_make("rtpL24pay", NULL);
+    self->udpsink         = gst_element_factory_make("udpsink", NULL);
 
-    if(!self->payloader || !self->udpsink)
+    if(!self->queue || !self->video_payloader || !self->audio_payloader || !self->udpsink)
```

Comment:
I see this all the time in C code, namely GStreamer code and I hate it.
If udpsink creation fails, all others are leaked. Same for payloader and the elements created before.
Also, using `!` to test for `!= nullptr` should be avoided.

You can leave everything as is but we need to add a task to solve these things.